common/src/test/java/com/revenuecat/purchases/common/offlineentitlements/PurchasedProductsFetcherTest.kt

```diff
@@ -0,0 +1,477 @@
+            productIdentifierAnnual to listOf(entitlements[1])
+        )
+
+        mockEntitlementMapping(mapOfEntitlements)
+
+        val activePurchase = stubStoreTransactionFromGooglePurchase(
+            productIds = listOf(productIdentifierMonthly),
+            purchaseTime = testDate.time
+        )
+        val purchaseRecordMonthly = stubStoreTransactionFromPurchaseHistoryRecord(
+            productIds = listOf(productIdentifierMonthly),
+            purchaseTime = testDate.time
+        )
+        val purchaseRecordAnnual = stubStoreTransactionFromPurchaseHistoryRecord(
+            productIds = listOf(productIdentifierAnnual),
+            purchaseTime = testDate.time
+        )
+
+        mockAllPurchases(purchaseRecordMonthly, purchaseRecordAnnual)
+        mockActivePurchases(activePurchase)
+
+        var receivedListOfPurchasedProducts: List<PurchasedProduct>? = null
+
+        fetcher.queryPurchasedProducts(
+            appUserID = "appUserID",
+            onSuccess = {
+                receivedListOfPurchasedProducts = it
+            },
+            unexpectedOnError
+        )
+
+        assertThat(receivedListOfPurchasedProducts).isNotNull
+        assertThat(receivedListOfPurchasedProducts!!.size).isEqualTo(2)
+
+        val monthlyProduct =
+            receivedListOfPurchasedProducts!!.first { it.productIdentifier == productIdentifierMonthly }
+        assertPurchasedProduct(
+            monthlyProduct,
+            purchaseRecordMonthly,
+            mapOfEntitlements,
+            isActive = true
+        )
+
+        val annualProduct = receivedListOfPurchasedProducts!!.first { it.productIdentifier == productIdentifierAnnual }
+        assertPurchasedProduct(
+            annualProduct,
+            purchaseRecordAnnual,
+            mapOfEntitlements,
+            isActive = false
+        )
+    }
+
+    private fun assertPurchasedProduct(
+        purchasedProduct: PurchasedProduct,
+        purchaseRecord: StoreTransaction,
+        productIdentifierToEntitlements: Map<String, List<String>>,
+        isActive: Boolean
+    ) {
+        assertThat(purchasedProduct.productIdentifier).isEqualTo(purchaseRecord.skus[0])
+        assertThat(purchasedProduct.storeTransaction).isEqualTo(purchaseRecord)
+        assertThat(purchasedProduct.isActive).isEqualTo(isActive)
+        assertThat(purchasedProduct.entitlements.size).isEqualTo(productIdentifierToEntitlements[purchasedProduct.productIdentifier]!!.size)
+        assertThat(purchasedProduct.entitlements).containsAll(productIdentifierToEntitlements[purchasedProduct.productIdentifier])
+        val expiresDate = if (isActive) testDatePlusOneDay else testDate
+        assertThat(purchasedProduct.expiresDate)
+            .withFailMessage("Expires date should be $expiresDate, but it was ${purchasedProduct.expiresDate}")
+            .isEqualTo(expiresDate)
+    }
+
+    private fun mockActivePurchases(vararg storeTransactions: StoreTransaction) {
```

Comment:
This is interesting, I would have simply passed a List here... But this should also work 👍

Reply:
yeah, I was first only passing one store transaction, and then I wanted to add multiple, so I just added vararg so I didn't have to change the other code.